### PR TITLE
Updated (GCI - 2018)

### PR DESCRIPTION
**Pull request by D3ezy27**

### CODE REVIEW

File: src/lib_ccaligner/commons.cpp [hunk 1]
```diff
@@ -20,24 +22,27 @@ void ms_to_srt_time(long int ms, int *hours, int *minutes, int *seconds, int *mi
...
+
 {
-    std::transform(strToConvert.begin(), strToConvert.end(), strToConvert.begin(), ::tolower);
+    transform(strToConvert.begin(), strToConvert.end(), strToConvert.begin(), tolower);
 
     return strToConvert;
 }
 
-bool AlignedData::addNewWord(const std::string& word, long int startTime, long int endTime, float conf)
+bool AlignedData_addNewWord(const string &word, long int startTime, long int endTime, float conf)
```
Comment (harrynull) [inline src/lib_ccaligner/commons.cpp]: `::` is part of the C++ grammar. Please do not change it.

File: src/lib_ccaligner/commons.cpp [hunk 2]
```diff
@@ -5,6 +5,8 @@
 */
 
 #include "commons.h"
+#include <string>
+using namespace std;
```
Comment (canihavesomecoffee) [inline src/lib_ccaligner/commons.cpp]: `using namespace std;` is considered bad practice. Please revert all changes related to this.

File: src/lib_ccaligner/commons.cpp [hunk 3]
```diff
@@ -5,6 +5,8 @@
 */
 
 #include "commons.h"
+#include <string>
```
Comment (canihavesomecoffee) [inline src/lib_ccaligner/commons.cpp]: Can you explain why this needs inclusion?

File: src/lib_ccaligner/commons.cpp [hunk 4]
```diff
@@ -20,24 +22,27 @@ void ms_to_srt_time(long int ms, int *hours, int *minutes, int *seconds, int *mi
...
-std::string extractFileName(const std::string& fileName)
+    string extractFileName(const string &fileName);
+
 {
     int lastIndex = fileName.find_last_of(".");
 
-    if(lastIndex == std::string::npos)
-        return fileName;    //if no extension is present, return complete filename
+    if(lastIndex == string npos)
+
```
Comment (canihavesomecoffee) [inline src/lib_ccaligner/commons.cpp]: Why the extra empty line here?

File: src/lib_ccaligner/commons.cpp [hunk 5]
```diff
@@ -20,24 +22,27 @@ void ms_to_srt_time(long int ms, int *hours, int *minutes, int *seconds, int *mi
...
-        return fileName;    //if no extension is present, return complete filename
+    if(lastIndex == string npos)
+        
+        return fileName;    //If no extension is present, return complete filename
     else
         return fileName.substr(0, lastIndex);
 }
 
-std::string stringToLower(std::string strToConvert)
+    string stringToLower(string strToConvert);
```
Comment (canihavesomecoffee) [inline src/lib_ccaligner/commons.cpp]: This ; terminates the method. That makes no sense.

File: src/lib_ccaligner/commons.cpp (hunk 1 above)
Comment (canihavesomecoffee) [reply src/lib_ccaligner/commons.cpp]: Indeed. `class_name::method_name` are extremely common practice to use.

**Comment by saurabhshri** [comment]
Sorry, closing the PR as none of the changes are apt for merge! Thanks @canihavesomecoffee and @harrynull for review. @D3ezy27 feel free to re-open this or send a new PR with relevant changes.

**Pull request closed without merging**